torchtitan/components/quantization/float8.py

```diff
@@ -26,9 +26,10 @@ def __init__(self, job_config: JobConfig, parallel_dims: ParallelDims):
         self.enabled = False
 
         float8_config: Float8 = job_config.float8
-        if not has_cuda_capability(8, 9):
+        if not has_cuda_capability(8, 9) and not float8_config.emulate:
```

Comment:
On sm < 89, we can't enable torch.compile with/without emulate, right? If so let's do
```suggestion
        if not has_cuda_capability(8, 9) and (job_config.training.compile or not float8_config.emulate):
```
Also it's a bit hard to read. A better way may be
```
if has_cuda_capability(8, 9) or (float8_config.emulate and not job_config.training.compile): pass
else: raise ValueError(...)
```